### PR TITLE
hw 29 is done

### PR DESCRIPTION
**Pull request by vladkuzya**

### CODE REVIEW

File: src/main/webapp/web.xml [hunk 1]
```diff
@@ -14,6 +14,16 @@
...
     </servlet-mapping>
 
+    <servlet>
+        <servlet-name>injectData</servlet-name>
+        <servlet-class>mate.academy.controllers.InjectDataController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>injectData</servlet-name>
+        <url-pattern>/injectData</url-pattern>
```
Comment (matviichuk92) [inline src/main/webapp/web.xml]: We shouldn't use camelSpace for urls. 
/inject-data will be better

**Review by matviichuk92 — APPROVED** [review]
great job!

File: src/main/java/mate/academy/model/Role.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package mate.academy.model;
+
+public class Role {
+    private Long id;
+    private RoleName roleName;
+
+    public Role(RoleName roleName) {
+        this.roleName = roleName;
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/model/Role.java]: let's create a static method of() for creating role which will take string as parameter and then make this constructor private

File: src/main/java/mate/academy/web/filter/AuthorizationFilter.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        protectedUrls.put("/products/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/all-admin", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/all-admin", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/shopping-carts/show", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/products/add", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/products/delete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/orders/complete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/orders/all-user", List.of(Role.RoleName.USER));
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/web/filter/AuthorizationFilter.java]:
```suggestion
        protectedUrls.put("user/orders", List.of(Role.RoleName.USER));
```
if you want to mark in url which urls accessible  to user and which are for admin, then let's put "admin" in those which are for admin(if you have the same url for user)

File: src/main/java/mate/academy/web/filter/AuthorizationFilter.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            .getInstance(UserService.class);
+    private Map<String, List<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/users/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/all-admin", List.of(Role.RoleName.ADMIN));
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/web/filter/AuthorizationFilter.java]:
```suggestion
        protectedUrls.put("admin/products/all", List.of(Role.RoleName.ADMIN));
```

File: src/main/java/mate/academy/web/filter/AuthorizationFilter.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        protectedUrls.put("/orders/all-admin", List.of(Role.RoleName.ADMIN));
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/web/filter/AuthorizationFilter.java]:
```suggestion
        protectedUrls.put("admin/orders/all", List.of(Role.RoleName.ADMIN));
```

File: src/main/webapp/WEB-INF/views/injectData.jsp [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<%--
+  Created by IntelliJ IDEA.
+  User: Влад
+  Date: 16.09.2020
+  Time: 22:58
+  To change this template use File | Settings | File Templates.
+--%>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/views/injectData.jsp]: remove these comments

File: src/main/webapp/web.xml (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/webapp/web.xml]: agree

Comment (vladkuzya) [reply src/main/webapp/web.xml]: done)

File: src/main/java/mate/academy/model/Role.java (hunk 2 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/model/Role.java]: Такой метод уже есть на 19 строчке)

File: src/main/webapp/WEB-INF/views/injectData.jsp (hunk 6 above)
Comment (vladkuzya) [reply src/main/webapp/WEB-INF/views/injectData.jsp]: done)

File: src/main/java/mate/academy/web/filter/AuthorizationFilter.java (hunk 5 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/web/filter/AuthorizationFilter.java]: done)

File: src/main/java/mate/academy/web/filter/AuthorizationFilter.java (hunk 4 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/web/filter/AuthorizationFilter.java]: done)

File: src/main/java/mate/academy/web/filter/AuthorizationFilter.java (hunk 3 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/web/filter/AuthorizationFilter.java]: done)

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**